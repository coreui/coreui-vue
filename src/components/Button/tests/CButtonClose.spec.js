import { mount } from '@vue/test-utils'
import Component from "../CButtonClose";

const ComponentName = 'CButtonClose'
const defaultWrapper = mount(Component)
const customWrapper = mount(Component, {
  attrs: {
    class: 'test2',
    disabled: 'disabled',
    'aria-label': 'test2',
    style: 'color:red'
  },
  slots: {
    default: 'ok'
  }
})

describe(ComponentName, () => {
  it('has a name', () => {
    expect(Component.name).toMatch(ComponentName)
  })
  it('renders correctly', () => {
    expect(defaultWrapper.element).toMatchSnapshot()
  })
  it('renders correctly', () => {
    expect(customWrapper.element).toMatchSnapshot()
  })
})
