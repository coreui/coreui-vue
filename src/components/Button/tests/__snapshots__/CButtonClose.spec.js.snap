// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CButtonClose renders correctly 1`] = `
<button
  aria-label="Close"
  class="close"
  type="button"
>
  Ã—
</button>
`;

exports[`CButtonClose renders correctly 2`] = `
<button
  aria-label="test2"
  class="test2"
  disabled="disabled"
  style="color:red"
  type="button"
>
  ok
</button>
`;
